src/main/java/Data.java

```diff
@@ -0,0 +1,84 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.IOException;
+import tasks.*;
+
+public class Data {
+    public static void createFile() throws IOException {
+        try {
+            if (new File("data").mkdirs()) {
+                File data = new File("data/data.txt");
+                data.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+    }
+
+    public static void saveData(ArrayList<Task> taskManager) throws IOException {
+        try {
+            createFile();
+        } catch (IOException e) {
+            System.out.println("An error occurred. Directory probably exists");
+        }
+
+        try {
+            FileWriter myWriter = new FileWriter("data/data.txt");
+            for (Task task : taskManager) {
+                if (task.getType().equals("T")) {
+                    myWriter.write(task.getType() + "%" + task.getStatusIcon() + "%" + task.getDescription() + '\n');
+                } else if (task.getType().equals("D")) {
+                    Deadline obj = (Deadline) task;
+                    myWriter.write(task.getType() + "%" + task.getStatusIcon() + "%" + task.getDescription() + "%" + obj.by + '\n');
+                } else if (task.getType().equals("E")) {
+                    Event obj = (Event) task;
+                    myWriter.write(task.getType() + "%" + task.getStatusIcon() + "%" + task.getDescription() + "%" + obj.start + "%" + obj.end + '\n');
```

Comment:
Java Coding Standard: Lines should be no longer than 120 chars. Consider using line wrapping to break such long lines of code down.